### PR TITLE
[DEV-2167] vector_agg: add support for snowflake avg function

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add support for snowflake AVG vector agg function.

We unskip the integration test to show that it works.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/session/snowflake.py [hunk 1]
```diff
@@ -311,8 +311,9 @@ def get_columns_schema_from_dataframe(dataframe: pd.DataFrame) -> list[tuple[str
                 db_type = "INT"
             elif (
                 dataframe.shape[0] > 0
-                and dataframe[colname].apply(lambda x: isinstance(x, list)).all()
+                and dataframe[colname].apply(lambda x: x is None or isinstance(x, list)).all()
```
Comment (gee-senbong) [inline featurebyte/session/snowflake.py:314]: nit: Should we use `np.isnan` here?

**Review by gee-senbong — APPROVED** [review]
LGTM

File: featurebyte/session/snowflake.py [hunk 2]
```diff
@@ -347,7 +348,7 @@ def sql_directory_name(self) -> str:
 
     @property
     def current_working_schema_version(self) -> int:
-        return 26
+        return 27
```
[the remark this answers is not in the document]
Comment (jevonyeoh) [reply featurebyte/session/snowflake.py:351]: oops, forgot to rm.

File: featurebyte/session/snowflake.py (hunk 1 above)
Comment (jevonyeoh) [reply featurebyte/session/snowflake.py:315]: hm seems like a bunch of errors are thrown (https://github.com/featurebyte/featurebyte/actions/runs/6071849693/job/16470734256?pr=1711) if i try to use that. will keep as is for now!

**Review by YS-L — APPROVED** [review]

**Pull request merged**